### PR TITLE
Fix ubuntu efi iso creation

### PR DESCRIPTION
**Pull request by Itxaka**
Fix grub.cfg location for ubuntu
Fix grub.efi name cleanup

### CODE REVIEW

File: pkg/action/build-iso.go [hunk 1]
```diff
@@ -183,6 +183,19 @@ func (b BuildISOAction) createEFI(rootdir string, isoDir string) error {
 		b.cfg.Logger.Errorf("Failed writing grub.cfg: %v", err)
 		return err
 	}
+	// Ubuntu efi searches for the grub.cfg file under /EFI/ubuntu/grub.cfg while we store it under /boot/grub2/grub.cfg
+	// workaround this by copying it there as well
+	// TODO: Should we symlink all the distros folders to our single grub.cfg?
```
Comment (jimmykarily) [inline pkg/action/build-iso.go:188]: If symlinks work for this, this sounds like a good idea

File: pkg/action/build-iso.go [hunk 2]
```diff
@@ -183,6 +183,19 @@ func (b BuildISOAction) createEFI(rootdir string, isoDir string) error {
...
+	err = utils.MkdirAll(b.cfg.Fs, filepath.Join(isoDir, "EFI/ubuntu/"), constants.DirPerm)
```
Comment (jimmykarily) [inline pkg/action/build-iso.go:189]: so every distro will have an `ubuntu` directory now? Maybe we should try the symlink idea instead. That fix would happen in the Dockerfiles right?

Comment (Itxaka) [reply pkg/action/build-iso.go:189]: Only for isos. The final system should not have that.

Comment (Itxaka) [reply pkg/action/build-iso.go:189]: on the install part, we will need to do the same for ubuntu, but in that point we know what distro is due to being able to read /etc/os-release so we can only trigger the same thing if we are on ubuntu.

Maybe we should do the same here.....

File: pkg/action/build-iso.go [hunk 3]
```diff
@@ -183,6 +184,27 @@ func (b BuildISOAction) createEFI(rootdir string, isoDir string) error {
 		b.cfg.Logger.Errorf("Failed writing grub.cfg: %v", err)
 		return err
 	}
+	// Ubuntu efi searches for the grub.cfg file under /EFI/ubuntu/grub.cfg while we store it under /boot/grub2/grub.cfg
+	// workaround this by copying it there as well
+	// read the os-release from the rootfs to know if we are creating a ubuntu based iso
+	FLAVOR, err := sdk.OSRelease("FLAVOR", filepath.Join(rootdir, "etc/os-release"))
```
Comment (jimmykarily) [inline pkg/action/build-iso.go:190]: [Minor] No reason to use upper cased variable names. Just `flavor` should be ok.

Comment (Itxaka) [reply pkg/action/build-iso.go:190]:
```
ERRO[2024-01-12T14:22:07Z] Failed preparing ISO's root tree: KAIROS_flavor key not found 
```

case sensitive!

Comment (Itxaka) [reply pkg/action/build-iso.go:190]: :facepalm:

Comment (jimmykarily) [reply pkg/action/build-iso.go:190]: sorry yes, I meant the golang variable, not the `os-release` one.

**Pull request merged**